lib/docker-toolbelt.coffee

```diff
@@ -79,9 +93,66 @@ Docker::imageRootDir = (image) ->
 					imageInfo.GraphDriver.Data.RootDir
 				when 'vfs'
 					path.join(dkroot, 'vfs/dir', destId)
+				when 'aufs'
+					path.join(dkroot, 'aufs/diff', destId)
 				else
 					throw new Error("Unsupported driver: #{dockerInfo.Driver}/")
 
+# Same as imageRootDir, but provides the full mounted rootfs for AUFS,
+# and has a disposer to unmount.
+Docker::imageRootDirMounted = (image) ->
+	docker = this
+	driver = null
+	Promise.all [
+		docker.infoAsync()
+		docker.versionAsync().get('Version')
+		docker.getImage(image).inspectAsync()
+	]
+	.spread (dockerInfo, dockerVersion, imageInfo) ->
+		driver = dockerInfo.Driver
+		dkroot = dockerInfo.DockerRootDir
+		imageId = imageInfo.Id
+		return docker.imageRootDir(image) if driver != 'aufs'
+		docker.aufsDiffPaths(image)
+		.then (layerDiffPaths) ->
+			branchesOption = 'br=' + layerDiffPaths.join('=ro:') + '=ro'
+			rootDir = path.join(dkroot, 'aufs/mnt', 'tmp' + imageId.split(':')[1])
+			fs.mkdirAsync(rootDir)
+			.then ->
+				execAsync("mount -t aufs -o 'noxino,ro,#{branchesOption}' none #{rootDir}")
+			.return(rootDir)
+	.disposer (rootDir) ->
+		return if driver != 'aufs'
```

Comment:
I personally prefer the reading of this as `driver isnt 'aufs'` but it's not a big deal